test_unstructured_ingest/test-ingest-pinecone.sh

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env bash
+
+set -e
+
+SCRIPT_DIR=$(dirname "$(realpath "$0")")
+cd "$SCRIPT_DIR"/.. || exit 1
+OUTPUT_FOLDER_NAME=s3-pinecone-dest
+OUTPUT_DIR=$SCRIPT_DIR/structured-output/$OUTPUT_FOLDER_NAME
+WORK_DIR=$SCRIPT_DIR/workdir/$OUTPUT_FOLDER_NAME
+DOWNLOAD_DIR=$SCRIPT_DIR/download/$OUTPUT_FOLDER_NAME
+
+if [ -z "$OPENAI_API_KEY" ] && [ -z "$PINECONE_API_KEY" ]; then
+   echo "Skipping Pinecone ingest test because neither OPENAI_API_KEY nor PINECONE_API_KEY env vars are set."
+   exit 0
+fi
+
+PINECONE_ENVIRONMENT="gcp-starter"
+PINECONE_INDEX="ingest-test"
+PINECONE_PROJECT_ID="bfa06d5"
+
+# shellcheck disable=SC1091
+source "$SCRIPT_DIR"/cleanup.sh
+function cleanup {
+
+  # Get response code to check if index exists
+  response_code=$(curl \
+    -s -o /dev/null \
+    -w "%{http_code}" \
+    --request GET \
+    --url https://controller.$PINECONE_ENVIRONMENT.pinecone.io/databases/$PINECONE_INDEX \
+    --header 'accept: application/json' \
+    --header "Api-Key: $PINECONE_API_KEY")
+
+  # Cleanup (delete) index if it exists
+  if [ "$response_code" == "200" ]; then
+    echo ""
+    echo "deleting index $PINECONE_INDEX"
+    curl --request DELETE \
+    "https://controller.$PINECONE_ENVIRONMENT.pinecone.io/databases/$PINECONE_INDEX" \
+    --header "Api-Key: $PINECONE_API_KEY" \
+    --header 'content-type: application/json'
+
+  else
+    echo "Index $PINECONE_INDEX does not exist, nothing to delete"
+  fi
+
+  # Local file cleanup
+  cleanup_dir "$WORK_DIR"
+  cleanup_dir "$OUTPUT_DIR"
+  if [ "$CI" == "true" ]; then
+    cleanup_dir "$DOWNLOAD_DIR"
+  fi
+}
+
+trap cleanup EXIT
+
+
+# Create index
+echo "Creating index $PINECONE_INDEX"
+
+response_code=$(curl \
+     -s -o /dev/null \
+     -w "%{http_code}" \
+     --request POST \
+     --url "https://controller.$PINECONE_ENVIRONMENT.pinecone.io/databases" \
+     --header "accept: text/plain" \
+     --header "content-type: application/json" \
+     --header "Api-Key: $PINECONE_API_KEY" \
+     --data '
+{
+  "name": "'"$PINECONE_INDEX"'",
+  "dimension": 1536,
+  "metric": "cosine",
+  "pods": 1,
+  "replicas": 1,
+  "pod_type": "p1.x1"
+}
+')
+
+
+if [ "$response_code" -lt 400 ]; then
+  echo "Index creation success: $response_code"
+else
+  echo "Index creation failure: $response_code"
+  exit 1
+fi
+
+PYTHONPATH=. ./unstructured/ingest/main.py \
+  s3 \
```

Comment:
To avoid network calls, I've been using the local source connector to generate the content to push up. Take a look at the tests I created in this PR: [1777](https://github.com/Unstructured-IO/unstructured/pull/1777)

Reply:
Agreed, addressed with [test and session creation updates](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/84e65e5389f05c57508311755c837d8d0b0815dc)